snowpack/assets/hmr-error-overlay.js

```diff
@@ -0,0 +1,950 @@
+    <p>
+        <span id="error-file-loc">Loading...</span>
+        <!--<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line></svg>-->
+    </p>
+    <hr><pre>Loading...</pre></div>
+    </div></div></div></div>
+</template>`;
+
+var wrapper = document.createElement('div');
+wrapper.innerHTML = ERROR_OVERLAY_TEMPLATE;
+document.body.appendChild(wrapper);
+
+class HmrErrorOverlay extends HTMLElement {
+  constructor({ title, errorMessage, fileLoc, errorStackTrace }) {
+    super();
+    this.title = title;
+    this.errorMessage = errorMessage;
+    this.fileLoc = fileLoc;
+    this.errorStackTrace = errorStackTrace;
+    this.sr = this.attachShadow({ mode: 'open' });
+    const template = document.getElementById('snowpack-error-overlay');
+    this.sr.appendChild(template.content.cloneNode(true));
+    this.close = this.close.bind(this);
+  }
+
+  connectedCallback() {
+    this.sr
+      .getElementById('close-button')
+      .addEventListener('click', this.close);
+    this.sr
+      .querySelector('[data-nextjs-dialog-backdrop]')
+      .addEventListener('click', this.close);
+
+    this.sr.getElementById(
+      'nextjs__container_errors_label',
+    ).innerText = this.title;
+    this.sr.getElementById(
+      'nextjs__container_errors_desc',
+    ).innerText = this.errorMessage;
+    if (this.fileLoc) {
+      this.sr.getElementById('error-file-loc').innerText = this.fileLoc;
+    } else {
+      this.sr.getElementById('error-file-loc').innerText = 'No source file.';
+    }
+    if (this.errorStackTrace) {
+      this.sr.querySelector('pre').innerText = this.errorStackTrace;
+    } else {
+      this.sr.querySelector('pre').style.display = 'none';
+    }
+  }
+
+  disconnectedCallback() {
+    this.sr
+      .getElementById('close-button')
+      .removeEventListener('click', this.close);
+    this.sr
+      .querySelector('[data-nextjs-dialog-backdrop]')
+      .removeEventListener('click', this.close);
+  }
+
+  close() {
+    this.parentNode.removeChild(this);
+  }
+
+  _watchEscape(event) {
+    if (event.key === 'Escape') {
+      this.close();
+    }
+  }
+}
+
+customElements.define('hmr-error-overlay', HmrErrorOverlay);
+// const BiggerImage = customElements.get('hmr-error-overlay');
+// const overlay = new BiggerImage('Build Error', '[esbuild] Could not build this file', 'src/index.jsx', new Error('AH').stack);
+// document.body.appendChild(overlay);
```

Comment:
Not a blocker, but what parts of this file were modified for Snowpack, if any? It looks like a copy–paste to me, but would love to know if anything needed to change to work.

Reply:
added a note to the code!